test/sqllogictest/advent-of-code/2023/aoc_1216.slt

```diff
@@ -266,74 +266,65 @@ Explained Query:
     cte l13 =
       Distinct project=[#0..=#3]
         Union
-          Distinct project=[#0..=#3]
+          Distinct project=[#0, #1, #3, #2]
```

Comment:
This looks like a `Distinct` that could be removed. It's possible it arrives after the pass is run, but worth diving in to and seeing if it is an oversight.

Reply:
The reason is that another term in the `Union` failed the non-negativity test. As such, there could be negative accumulations, and the `Distinct` is load-bearing because the cardinality of positive records that might cancel the negative ones must be preserved. Inspection suggests that actually our non-negativity analysis is weak, but .. going deep on that seems potentially out of scope.